### ISSUE TITLE
perform computation involving multiple fields

### ISSUE THREAD

**Issue by HuStmpHrrr** [joaotavora/yasnippet#898]
I searched around and found this question: https://stackoverflow.com/questions/23413847/how-to-add-values-of-two-yasnippet-fields
I am pretty much trying to achieve the same thing and the question is a bit old already so I am wondering if it's feasible now.

A good use case for this would be to consider defining a template for classes in python:

```
class $1($2):
    def __init__(self, $3):
        ${2:$(if (= $2 "object") "" "super($1, self).__init__($3)")}
```

consider the third line of code. Ideally, if a class is just an object, then there is no need to call `super`; however, if a class inherits from some other classes, then it's useful to generate a super call. this then has to be implemented by knowing all three information: the class name, the parent name, and the fields. If it's possible, then it opens to door to very flexible ways of defining templates. I am wondering if it's possible to do such operation now.

**Comment by joaotavora** [comment]
I believe that what you are looking for is the poorly documented, [but not entirely undocumented](http://joaotavora.github.io/yasnippet/snippet-reference.html#yas-field-value), function `yas-field-value` that you can use in (mirror and field transformation expressions)[http://joaotavora.github.io/yasnippet/snippet-development.html#sec-3-6]. 

This function takes a number of a field as an argument and returns the string of text contained by that field. So a totally untested thing is:

```
class $1($2):
    def __init__(self, $3):
        ${2:$(if (string-equal (yas-field-value 2) "object")
                 ""
                 (concat "super(" 
                         (yas-field-value 1) 
                         ", self).__init__(" 
                         (yas-field-value 3) 
                         ")"))}
```

**Comment by npostavs** [comment]
Huh. I was not aware of `yas-field-value`. The docstring seems to trail off in a typo, I fixed it up in #903.
